src/components/figures/figure_container/figure_container.ts

```diff
@@ -117,14 +119,26 @@ export class FiguresContainer extends Component<Props, SpreadsheetChildEnv> {
       // new rendering
       this.render();
     });
+    onWillUpdateProps(() => {
+      const sheetId = this.env.model.getters.getActiveSheetId();
+      if (this.dnd.figId && !this.env.model.getters.getFigure(sheetId, this.dnd.figId)) {
+        this.dnd.cancelDnd?.();
```

Comment:
in 16.0, we cannot use optional chaining :/
We are bound to es10 :/

https://github.com/odoo/odoo/blob/6c3c7562258e8996d7ad43d6f8ae766799c68b8f/odoo/addons/test_lint/tests/test_ecmascript.py#L14